tests/Unit/ProtoTests/Basic/out/src/Client/BasicClient.php

```diff
@@ -208,4 +210,30 @@ public function methodWithArgs(RequestWithArgs $request, array $callOptions = []
     {
         return $this->startApiCall('MethodWithArgs', $request, $callOptions)->wait();
     }
+
+    /** Configure the gapic configuration to use a service emulator. */
+    private function getDefaultEmulatorConfig(): array
```

Comment:
What if we moved the `$emulatorHost` parameter up into the method signature? Then we could move the entire method into `GapicClientTrait` instead of generating it every time:
```suggestion
    private function getDefaultEmulatorConfig(string $emulatorHost): array
```

Reply:
I am open to the above mentioned idea although I like this approach.

I am thinking of going with the current implemented approach for now (as its a `private` method) and if we get a similar requirement from other products, we will look into multiple options and take the decision?

Reply:
Yes that sounds great. I agree what we have here is better, but wanted to explore every option.

Reply:
Thank you!